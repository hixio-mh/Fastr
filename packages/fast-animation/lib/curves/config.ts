import { IBezierCurve } from "./index";

export const linear: IBezierCurve = [
    [0, 0],
    [1, 1]
];
export const easeOut: IBezierCurve = [
    [0, 0],
    [0.58, 1]
];
export const easeOutSmooth: IBezierCurve = [
    [0, 0.35],
    [0.15, 1]
];
export const easeIn: IBezierCurve = [
    [0.25, 0.1],
    [0.25, 1]
];
export const drillIn: IBezierCurve = [
    [0.17, 0.17],
    [0, 1]
];
export const backToApp: IBezierCurve = [
    [0.5, 0],
    [0.6, 1]
];
export const appToApp: IBezierCurve = [
    [0.5, 0],
    [1, 0.9]
];
export const fastIn: IBezierCurve = [
    [0.1, 0.9],
    [0.2, 1]
];
export const fastOut: IBezierCurve = [
    [0.9, 0.1],
    [1, 0.2]
];
export const fastInOut: IBezierCurve = [
    [0.8, 0],
    [0.2, 1]
];
export const exponential: IBezierCurve = [
    [0.1, 0.25],
    [0.75, 0.9]
];
export const fastInFortySevenPercent: IBezierCurve = [
    [0.11, 0.5],
    [0.24, 0.96]
];
export const exponentialReversed: IBezierCurve = [
    [0.25, 0.1],
    [0.9, 0.75]
];
export const navPane: IBezierCurve = [
    [0.1, 0.7],
    [0.1, 1]
];
