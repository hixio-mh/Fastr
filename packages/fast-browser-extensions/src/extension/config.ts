/**
 * Define configuration values
 */
const validDomains: string[] = ["*://*.fluentweb.com/*"];
const extensionId: string = "jhhigejkemaekdcempcigaapebobmimg";

/**
 * Define event names
 */
export const menuItemClickEvent: string = "FW_EXTENSION::MENU_ITEM_CLICK";

/**
 * Non-production environments
 */
if (process.env.NODE_ENV !== "production") {
    validDomains.push("<all_urls>");
}

/**
 * Assign exports
 */
export { validDomains, extensionId };
