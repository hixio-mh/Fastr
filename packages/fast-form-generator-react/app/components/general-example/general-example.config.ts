import {
    IFormComponentMappingToPropertyNamesProps
} from "../../../src/form/form.props";

export default {
    alignHorizontal: [
        "alignHorizontal"
    ],
    alignVertical: [
        "alignVertical"
    ],
    theme: [
        "theme"
    ]
} as IFormComponentMappingToPropertyNamesProps;
