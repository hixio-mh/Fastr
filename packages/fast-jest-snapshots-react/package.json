{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "sideEffects": false,
  "version": "1.8.0",
  "author": {
    "name": "Microsoft",
    "url": "https://discord.gg/FcSNfg4"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.8.0",
    "@types/jest": "^22.2.2",
    "@types/react": "^16.3.6",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
