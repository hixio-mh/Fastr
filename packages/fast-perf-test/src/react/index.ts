import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
