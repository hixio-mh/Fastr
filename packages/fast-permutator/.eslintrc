{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "classes": true
        }
    },
    "rules": {
        "semi": 2
    }
}