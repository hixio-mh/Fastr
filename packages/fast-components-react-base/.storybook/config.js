import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
// Import from dist cause storybook throws up if we import the named import from the index
import { theme } from "@microsoft/fast-storybook-presets";

addParameters({
    options: {
        theme,
    },
});
