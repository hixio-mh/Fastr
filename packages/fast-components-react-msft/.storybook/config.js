import React from "react";
import { configure, addParameters, addDecorator } from "@storybook/react";
import { create, Global } from "@storybook/theming";
import { addReadme } from "storybook-readme";

// Import from dist cause storybook throws up if we import the named import from the index
import theme from "@microsoft/fast-storybook-presets/dist/theme";

addDecorator(addReadme);
addParameters({
    options: {
        theme,
    },
});
