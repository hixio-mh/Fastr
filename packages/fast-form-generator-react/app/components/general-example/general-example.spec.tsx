import * as React from "react";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import { SnapshotTestSuite } from "@microsoft/fast-jest-snapshots-react";
import { getExample } from "@microsoft/fast-permutator";
import Form, { FormProps } from "../../../src/form";
import * as generalExampleSchema from "./general-example";
import config from "./general-example.config";

const name: string = "general-example";

const exampleData: FormProps = {
    schema: generalExampleSchema,
    data: getExample(generalExampleSchema),
    /* tslint:disable-next-line */
    onChange: (data: any): void => {},
    componentMappingToPropertyNames: config,
};

const examples: SnapshotTestSuite<FormProps> = {
    name,
    component: Form,
    data: [exampleData],
};

describe(name, () => {
    generateSnapshots(examples);
});
