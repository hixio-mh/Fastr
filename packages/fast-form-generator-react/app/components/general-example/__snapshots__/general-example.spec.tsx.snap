// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general-example general-example: 0 1`] = `
<div
  className={null}
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <div>
        <div />
      </div>
    </div>
  </form>
</div>
`;
